### ISSUE TITLE
["BUG"] After upgrading to x7 today, the cursor of the ai dialog box is no longer displayed. It used to be a flashing red cursor. | [mod=chat] [v0.4.4]

### ISSUE THREAD

**Issue by QianChenglong** [x-cmd/x-cmd#58]
**Describe the bug**

cursor is not show in ai chat.

**System (please complete the following information):**

chenglong@qianchenglongdeMacBook-Pro-2.local 
-------------------------------------------- 
OS: macOS 14.5 23F79 arm64 
Host: MacBookPro18,1 
Kernel: 23.5.0 
Uptime: 39 days, 5 hours, 42 mins 
Shell: fish 3.7.1 
Resolution: 1728x1117@2x, 2560x1440@2x 
DE: Aqua 
Terminal: zellij 
GPU: Apple M1 Pro 
Memory: 6469MiB / 32768MiB 

**Screenshots**
old：
<img width="908" alt="image" src="https://github.com/user-attachments/assets/7caf0b12-3a98-4305-8962-03a067338b71">

new：
<img width="345" alt="image" src="https://github.com/user-attachments/assets/ceede035-4c43-4f30-9349-5c3b97aacdbb">

**To Reproduce**
1. Start up service: '...'
2. Type command: '...'

**Expected behavior**

**Additional context**

**Comment by edwinjhlee** [comment]
OK. I will take a look on it.

I haven't been able to reproduce the cursor problem you described. 
I'll ask my colleagues to see if they can replicate it.

**Comment by QianChenglong** [comment]
same problem using in alacritty .
<img width="638" alt="image" src="https://github.com/user-attachments/assets/9ace4b82-c0fc-4053-89e6-92e6ce97db61">

**Comment by QianChenglong** [comment]
how can i rollback to last version?

**Comment by iandol** [comment]
I think `x upgrade latest` gets you back, or just run the standard `eval "$(curl https://get.x-cmd.com)"` installer...

**Comment by QianChenglong** [comment]
> I think `x upgrade latest` gets you back, or just run the standard `eval "$(curl https://get.x-cmd.com)"` installer...

last version is not latest version

**Comment by Zhengqbbb** [comment]
The issue can be reproduced using other terminals ?
Could you also provide the version number of gum by running `x gum -v` ?

I am using zellij, iterm2, and apple terminal, all of which are working fine.

![CleanShot 2024-07-25 at 18 02 51@2x](https://github.com/user-attachments/assets/d14128d4-e884-42b3-9c64-9e221f3c756b)

**Comment by QianChenglong** [comment]
<img width="588" alt="image" src="https://github.com/user-attachments/assets/a62e1a9b-c242-47c8-b59b-db4b87eaafa2">

**Comment by Zhengqbbb** [comment]
> <img alt="image" width="588" src="https://private-user-images.githubusercontent.com/1881126/352084448-a62e1a9b-c242-47c8-b59b-db4b87eaafa2.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjE5MDkxNzEsIm5iZiI6MTcyMTkwODg3MSwicGF0aCI6Ii8xODgxMTI2LzM1MjA4NDQ0OC1hNjJlMWE5Yi1jMjQyLTQ3YzgtYjU5Yi1kYjRiODdlYWFmYTIucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDcyNSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA3MjVUMTIwMTExWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9OGM1NzFmZWFkODY2ZTI4NGNiYTlmNTZmYTZlYWUyYTUxYTJlYTIyZGI3NTExZGJkMDY4NjY2NTZmYjc4N2Q3MSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.Sq0xFN4B2nkzf1Gb2IP6hhnq98Qc8wAcwqGlJfQ5nDo">

If this issue occurs under this command, it is considered a defect in the upstream package, and we will find a way to replace it. 
So please check if this problem arises when running the command.


```sh
x gum input --prompt=">>> " --width 0 --placeholder="Enter to end the input, Ctrl-C to exit, /h for help"
```

**Comment by QianChenglong** [comment]
yes, still no cursor.
<img width="1074" alt="image" src="https://github.com/user-attachments/assets/7d8199cd-8024-48a1-a80d-2cd00b37b3f7">

**Comment by qiakai** [comment]
This might be a bug in gum v0.14.1, previously x-cmd used gum v0.13.0

**Comment by QianChenglong** [comment]
How to temporarily specify the version to avoid this problem, which greatly affects the use

**Comment by edwinjhlee** [comment]
I decide to give up `gum input.` 

Instead we try fzf as a one-liner editor. ( Also with history supports )
It switchs to `gum write` if necessary.

This solution requires further refinement. We welcome your feedback and suggestions. We will implement necessary adjustments in this week.

**Comment by QianChenglong** [comment]
I'd love to try this approach

**Comment by edwinjhlee** [comment]
Just like https://github.com/x-cmd/x-cmd/issues/35 

The issue with gum input has been addressed by switching to fzf.

We've implemented fzf as a replacement, providing a more robust solution. Please take a look ~

**Comment by edwinjhlee** [comment]
This issue should be resolved in v0.4.4.  
Please reopen this issue if the problem persists.
